engine/table/src/main/java/io/deephaven/engine/table/impl/select/CharRangeFilter.java

```diff
@@ -163,10 +164,19 @@ private long bound(RowSet selection, boolean usePrev, ColumnSource<Character> lo
 
     @Override
     public boolean overlaps(
-            @NotNull final Object lower,
-            @NotNull final Object upper,
+            @Nullable final Object lower,
+            @Nullable final Object upper,
             final boolean lowerInclusive,
             final boolean upperInclusive) {
+        // Validate the input bounds.
+        final int c0 = CompareUtils.compare(lower, upper);
+        if (c0 > 0) {
+            throw new IllegalArgumentException("Lower bound must not be greater than upper bound, found: "
+                    + lower + " > " + upper);
+        } else if (c0 == 0 && (!lowerInclusive || !upperInclusive)) {
+            throw new IllegalArgumentException("Lower and upper bounds must be inclusive when equal, found: "
+                    + lower + " == " + upper);
+        }
```

Comment:
Is throwing an exception the correct thing to do? I would argue that the definition of overlaps is still defined , even when lower == upper and under all inclusivity combinations. The answer would be `false`, does not overlap for all input.